internal/cmd/argus/instance/create/create_test.go

```diff
@@ -0,0 +1,315 @@
+				projectIdFlag:    testProjectId,
+				planIdFlag:       testPlanId,
+				instanceNameFlag: "",
+			},
+			isValid: true,
+			expectedModel: &inputModel{
+				GlobalFlagModel: &globalflags.GlobalFlagModel{
+					ProjectId: testProjectId,
+				},
+				PlanId:       utils.Ptr(testPlanId),
+				InstanceName: utils.Ptr(""),
+			},
+		},
+		{
+			description: "project id missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, projectIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 1",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 2",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+		{
+			description: "invalid with plan ID and plan name",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[planNameFlag] = "plan-name"
+			}),
+			isValid: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			cmd := NewCmd()
+			err := globalflags.Configure(cmd.Flags())
+			if err != nil {
+				t.Fatalf("configure global flags: %v", err)
+			}
+
+			for flag, value := range tt.flagValues {
+				err := cmd.Flags().Set(flag, value)
+				if err != nil {
+					if !tt.isValid {
+						return
+					}
+					t.Fatalf("setting flag --%s=%s: %v", flag, value, err)
+				}
+			}
+
+			err = cmd.ValidateRequiredFlags()
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error validating flags: %v", err)
+			}
+
+			model, err := parseInput(cmd)
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error parsing flags: %v", err)
+			}
+
+			if !tt.isValid {
+				t.Fatalf("did not fail on invalid input")
+			}
+			diff := cmp.Diff(model, tt.expectedModel)
+			if diff != "" {
+				t.Fatalf("Data does not match: %s", diff)
+			}
+		})
+	}
+}
+
+func TestBuildRequest(t *testing.T) {
+	tests := []struct {
+		description       string
+		model             *inputModel
+		expectedRequest   argus.ApiCreateInstanceRequest
+		getOfferingsFails bool
+		getPlansReponse   *argus.PlansResponse
+		isValid           bool
+	}{
+		{
+			description:     "base",
+			model:           fixtureInputModel(),
+			expectedRequest: fixtureRequest(),
+			getPlansReponse: &argus.PlansResponse{
+				Plans: &[]argus.Plan{
+					{},
+				},
+			},
+		},
+		{
+			description: "use plan name",
+			model: fixtureInputModel(
+				func(model *inputModel) {
+					model.PlanId = nil
+					model.PlanName = "example-plan-name"
+				},
+			),
+			expectedRequest: fixtureRequest(),
+			getPlansReponse: &argus.PlansResponse{
+				Plans: &[]argus.Plan{
+					{
+						Name: utils.Ptr("example-plan-name"),
+						Id:   utils.Ptr(testPlanId),
+					},
+				},
+			},
+		},
```

Comment:
These test cases do not `isValid` defined (it defaults to false). Are they suppose to be invalid? Either way I think it's better to explicitly set the field